src/JustEat.StatsD/Buffered/Buffer.cs

```diff
@@ -2,15 +2,18 @@
 
 namespace JustEat.StatsD.Buffered
 {
-    internal ref struct Buffer
+    /// <summary>
+    /// A class representing the buffer where the StatsD message is written.
+    /// </summary>
+    public ref struct Buffer
     {
-        public Buffer(Span<byte> source)
+        internal Buffer(Span<byte> source)
```

Comment:
Would this type actually be usable by a custom implementation of the tags formatter that isn't within the StatsD assembly itself if all the members of the buffer are internal?

Reply:
I have modified the interface IStatsDPublisher to avoid exposing the Buffer struct. I will review the performance of the different providers but I think that the extension method TryWriteUtf8String has to manage null string or empty strings.